crm_exception/models/crm_lead.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+from odoo.osv import expression
+
+
+class CrmLead(models.Model):
+    _inherit = ["crm.lead", "base.exception"]
+    _name = "crm.lead"
+    _order = "main_exception_id asc, name desc"
+
+    def write(self, vals):
+        result = super().write(vals)
+        # To avoid a recursive call, write()
+        if "exception_ids" in vals:
+            return result
+        self._check_exception()
+        return result
+
+    @api.model
+    def _reverse_field(self):
+        return "crm_lead_ids"
+
+    def _rule_domain(self):
+        base_rule_domain = super()._rule_domain()
+        if self.stage_id:
+            rule_domain = expression.AND(
+                [
+                    base_rule_domain,
+                    [
+                        "|",
+                        ("stage_ids", "in", tuple(self.stage_id.ids)),
+                        ("stage_ids", "=", False),
+                    ],
+                ]
+            )
+            return rule_domain
+        return base_rule_domain
```

Comment:
```suggestion
        rule_domain = super()._rule_domain()
        if self.stage_id:
            rule_domain = expression.AND(
                [
                    rule_domain,
                    [
                        "|",
                        ("stage_ids", "in", self.stage_id.ids),
                        ("stage_ids", "=", False),
                    ],
                ]
            )
        return rule_domain
```

Any reason to use tuple here? `tuple(self.stage_id.ids)`

Reply:
If we don't convert to tuple in here, cache lookeup error(`TypeError: unhashable type: 'list'`) will occur from this.
https://github.com/OCA/server-tools/blob/590cf43c30698fc8dd2fca306fe5bb6444ecb903/base_exception/models/exception_rule.py#L81C21-L81C49